kartothek/io/testing/index.py

```diff
@@ -183,3 +185,34 @@ def test_empty_partitions(store_factory, metadata_version, bound_build_dataset_i
     # Assert indices are properly created
     dataset_factory = DatasetFactory(dataset_uuid, store_factory, load_all_indices=True)
     assert_index_dct_equal(expected, dataset_factory.indices["p"].index_dct)
+
+
+def test_build_indices_doesnot_drop_metadata(
+    store_factory, bound_build_dataset_indices
+):
+    dataset_uuid = "dataset_uuid"
+    partitions = [
+        {"label": "cluster_1", "data": [("core", pd.DataFrame({"p": [1, 2]}))]},
+        {"label": "cluster_2", "data": [("core", pd.DataFrame({"p": [2, 3]}))]},
+    ]
+    with freeze_time(TIME_TO_FREEZE_ISO):
```

Comment:
IIRC, we had some issues with the `freeze_time` approach in the past, which is why almost no test nowadays uses it. I think this test can be re-written without using `freeze_time`, simply by not checking a value for `metadata["creation_time"]`. This would not only drop the dependency on `freezegun` here, but also make the test clearer.

Reply:
Makes sense. Removed `freeze_time` and pushed the changes.